models/ed/R/write.configs.ed.R

```diff
@@ -443,18 +443,22 @@ remove.config.ED2 <- function(main.outdir = settings$outdir, settings) {
 #' @author David LeBauer, Shawn Serbin, Carl Davidson, Alexey Shiklomanov
 write.config.xml.ED2 <- function(settings, trait.values, defaults = settings$constants) {
 
-  ## Find history file TODO this should come from the database
-  ed2_package_data <- data(package="PEcAn.ED2", envir = environment())
-  histfile <- paste0("history.r", settings$model$revision) # set history file name to look for in ed2_package_data
-  if (histfile %in% ed2_package_data$results[, "Item"]) {
+  # TODO this should come from the database
+  
+  if(settings$model$revision %in% c("46", "81", "82", "85", "git")) {
+    histfile <- paste0("history.r", settings$model$revision)
     PEcAn.logger::logger.debug(paste0("--- Using ED2 History File: ", histfile))
-    data(list=histfile, package = 'PEcAn.ED2', envir = environment())
-    edhistory <- get(histfile)
+    edhistory <-
```

Comment:
This solution doesn't seem scalable -- if anyone adds a new ED version to the database they need to know to make manual modifications to this function.

Reply:
Yeah, that makes sense.  I'll see if I can make this more "programatic".  For what its worth, there aren't history files for every version of ED2 that there's an ED2IN file for, so it's already not scaling.  It's not clear to me how to update the history file for a new ED2 version.

Reply:
Ok, tried a different strategy and added some comments for why this code is there.